### ISSUE TITLE
Timex.format! may defer for the same input (integer vs. float)

### ISSUE THREAD

**Issue by PragTob** [bitwalker/timex#176]
First thanks for timex and especially for the effort to have the calendar types in Elixir 1.3!

(timex version 2.1.4, also checked on master and happens there ass well)

### Steps to reproduce

```
iex(2)> time = %Timex.DateTime{ calendar: :gregorian, year: 2016, month: 2, day: 14, hour: 12, minute: 34, second: 00, millisecond: 0.0, timezone: %Timex.TimezoneInfo{abbreviation: "UTC", full_name: "UTC"}}
#<DateTime(2016-02-14T12:34:00Z)>
iex(3)> Timex.format! time, "{ISOz}"
"2016-02-14T12:34:00.000Z"
iex(4)> time = %Timex.DateTime{ calendar: :gregorian, year: 2016, month: 2, day: 14, hour: 12, minute: 34, second: 00, millisecond: 0, timezone: %Timex.TimezoneInfo{abbreviation: "UTC", full_name: "UTC"}}  
#<DateTime(2016-02-14T12:34:00Z)>
iex(5)> Timex.format! time, "{ISOz}"                                                                                                                      "2016-02-14T12:34:00Z"  
```

### Description of issue

I'd expect the output to be the same as the input only differs by millisecond being 0 vs 0.0. E.g. it should be one of "2016-02-14T12:34:00.000Z" or "2016-02-14T12:34:00Z" all the time imo and not one time the one thing, one time the other. Not sure what the ISO says (I think the milliseconds are optional) but it might be nice to consistently show them.

How did it bite us? Pattern matching in tests. When the values come from the database they would have the milliseconds, otherwise not.

Happy to do this/help with it if it is a desired feature.

**Comment by bitwalker** [comment]
This should now be fixed in 1.3. Could you check with master?

**Comment by bitwalker** [comment]
Confirmed this is fixed in 3.0

**Comment by PragTob** [comment]
Argh sorry forgot the check :/ thanks for doing and fixing this!
